consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -519,46 +520,7 @@ private void forwardWorldStateTo(final BlockHeader newHead) {
                     newHead.getHash()));
   }
 
-  @Override
-  public boolean latestValidAncestorDescendsFromTerminal(final BlockHeader blockHeader) {
-    if (blockHeader.getNumber() <= 1L) {
-      // parent is a genesis block, check for merge-at-genesis
-      var blockchain = protocolContext.getBlockchain();
-
-      return blockchain
-          .getTotalDifficultyByHash(blockHeader.getBlockHash())
-          .map(Optional::of)
-          .orElse(blockchain.getTotalDifficultyByHash(blockHeader.getParentHash()))
-          .filter(
-              currDiff -> currDiff.greaterOrEqualThan(mergeContext.getTerminalTotalDifficulty()))
-          .isPresent();
-    }
-
-    Optional<Hash> validAncestorHash = this.getLatestValidAncestor(blockHeader);
-    if (validAncestorHash.isPresent()) {
-      final Optional<BlockHeader> maybeFinalized = mergeContext.getFinalized();
-      if (maybeFinalized.isPresent()) {
-        return isDescendantOf(maybeFinalized.get(), blockHeader);
-      } else {
-        Optional<BlockHeader> terminalBlockHeader = mergeContext.getTerminalPoWBlock();
-        if (terminalBlockHeader.isPresent()) {
-          return isDescendantOf(terminalBlockHeader.get(), blockHeader);
-        } else {
-          if (isTerminalProofOfWorkBlock(blockHeader, protocolContext)
-              || ancestorIsValidTerminalProofOfWork(blockHeader)) {
```

Comment:
FYI ancestorIsValidTerminalProofOfWork is dead code if we drop latestValidAncestorDescendsFromTerminal